app/components/admin/datetime_component.html.erb

```diff
@@ -0,0 +1,64 @@
+<div class="gem-c-date-input govuk-date-input govuk-datetime <%= 'govuk-form-group--error' if errors_for_input(object.errors, field_name).present? %>">
+  <h3 class="govuk-heading-s govuk-!-margin-bottom-2 govuk-!-margin-top-4">Date</h3>
+
+  <div class="gem-c-hint govuk-hint govuk-!-margin-bottom-2">
+    For example, 01 August 2022
+  </div>
+
+  <% if errors_for_input(object.errors, field_name).present? %>
```

Comment:
should this errors_for_input check be something in the component code itself. It seems to do processing in the helper so seems like something that could be helpfully memo'd in the component.